### DISCUSSION TITLE
tweepy.errors.Unauthorized: 401 Unauthorized 89 - Invalid or expired token.

### DISCUSSION THREAD

**Discussion by ParkJunYeop** [tweepy/tweepy#1961] [Questions] [answered]
I have an API version2 of the following keys:
consumer_key,consumer_secret_key,access_token,access_token_secret

and I'm having this error:
tweepy.errors.Unauthorized: 401 Unauthorized
89 - Invalid or expired token.

I actually made system time to UTF. And I don't know why this is happening!
Do I have to do something more for the time?
Could you please help me? 

The following code is what I'm currently working on

auth = tweepy.OAuth1UserHandler(consumer_key,consumer_secret_key)

auth.set_access_token(access_token,access_token_secret)

api = tweepy.API(auth)
print(api)
def twitter_search(outfilename, keyword):
    result = []

    for i in range(1,4):
        tweets = api.search_tweets(keyword)
        result = []
        for tweet in tweets:
            result.append([tweet.created_at, tweet.id_str, tweet.text, tweet.favorite_count,tweet.retweet_count])
        result = sorted(result, key=lambda x:-x[-2]) 
    df = pd.DataFrame(result, columns = ['time','id','text','likey','retweet'])

    print(len(result)) 
    print(df) 

    #df to csv
    df.to_csv(outfilename, encoding = 'utf-8-sig')

if __name__=="__main__":
    outfilename = 'tweet_search.csv'
    keyword = '~~~~' 
    twitter_search(outfilename, keyword)

**Comment by Harmon758** [comment, marked as answer]
Double check that your credentials are correct. Regenerate them if necessary.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Comment by ParkJunYeop** [comment]
I actually regenerated several times already but it still gives me the same error.
But using Bearer token was fine. It worked!

And it says I already pulled about 10,000 tweets even though I wasn't able to do it even once...
Is that because of the bearer token that I used before?

Does selecting Read and write and Direct message from User authentication settings matters? I chose that...
Not the Read and write one.

**Reply by AKWA-GAIUS** [reply]
i am having a similar problem. i dont understand what youve said

**Reply by benjamin32561** [reply]
I am using the API Key and Secret and it does not work, how did you solve it?

**Reply by jerryola1** [reply]
He said using Bearer token. I am also have same issue right now.

**Comment by Harmon758** [comment]
It's almost certainly an issue with the actual credentials or how you're passing them.
Bearer tokens are used with Twitter API v2 and shouldn't be relevant to `API`, Tweepy's interface for Twitter API v1.1.

I'm not sure where you're seeing that, but that shouldn't be relevant to this issue.

No, the error indicates your credentials are invalid.

**Comment by ParkJunYeop** [comment]
Then should I use API v1.1 consumer_key,consumer_secret_key,access_token,access_token_secret?

**Comment by Harmon758** [comment]
`API`, Tweepy's interface for Twitter API v1.1, requires those credentials to authenticate with user context, yes.

**Comment by ParkJunYeop** [comment]
But when I typed in the tokens, I get this message even though I had already upgraded to 'elevated' level.

>>>>If you need access to this endpoint, you'll need to apply for Elevated access via the Developer Portal

,which is why I tried to use API v2

**Comment by Harmon758** [comment]
Assuming you mean it says Tweepy API v1.1 requires Elevated access, this is true, as the error you're likely seeing indicates.
https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api

That's unrelated to this issue though.
